### PR TITLE
Clock out with your glock out: Plexagon Crew Self Serve

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request

Adds a new PDA app, Plexagon Self Serve. This app will be a crew member's centralised application for HR related functions.

- Allows crew to clock in/out wherever they are
- Instead of sending people's restricted gear to cryo, it's instead placed into a patent pending ~~SHAME BOX~~ lockbox.
- The lockbox is unlocked and items returned once they've clocked back in.

Future PRs I'd like to add things like paycheque and other crew related HR stuff.

## Why It's Good For The Game

- A serial observer got mad I didn't clock out before ERP once
- Pain point: Having to actually walk across station to the time clock
- Pain point: Losing your items on clock out, then you need the captain or HoP to get them back on clock in.

## Changelog

:cl: LT3
refactor: Crew HR functions in a new PDA app, Plexagon Self Serve
refactor: Clock in and out from your PDA
qol: Job items locked away while clocked out instead of being warped to cryo; returned on clock in
/:cl:

### CODE REVIEW

**Comment by Iamgoofball** [comment]
>Instead of sending people's restricted gear to cryo, it's instead placed into a patent pending SHAME BOX lockbox.

Please just beam it into the head of staff office for the relevant department

**Comment by lessthnthree** [comment]
> > Instead of sending people's restricted gear to cryo, it's instead placed into a patent pending SHAME BOX lockbox.
> 
> Please just beam it into the head of staff office for the relevant department and change Cryogenics's wallmounted system to do the same; that'll let the head of staff distribute it to any new hires

It's drafted because yes, I will be converting the wall mounted time clock to use the name NTos software and functionality.

**Comment by Iamgoofball** [comment]
>Wouldnt it make the gear unable to be retrieved in cases like low-pop hours when there's barely any command members?

it's easier to break into a head of staff office and get the equipment off the floor than it is to get bwoinked for touching the cryogenics console or if it's stuffed into a lockbox that's on the floor of a dorm room hosting an enterprise resource planning session

**Comment by Iamgoofball** [comment]
also, I think if we stat gathered how many people clock out and then clock back in, you'd find that basically nobody clocks in

this is something you can actually measure with the logs but you'd have to get your host to agree to let you grep all the logs and do some data analysis

**Comment by lessthnthree** [comment]
I'm not entirely sold on putting the lockbox somewhere, but if we did go that path, I'd rather it be like a department's storage area. Both for the sake of employee retrieval, and I guess people can steal the lockbox if they want for some reason. 

I don't see what benefit it gives other than a chance for petty crime, and in the process eliminates half of the benefit this PR is meant to address, that you can clock in/out and have your gear WITHOUT having to run somewhere, be it a time clock or the CMO's office.

My original concept was the lock box would be inaccessible and podded down to you when you clock back in, so in that situation you wouldn't be stealing anything anyways.

**Comment by SomeRandomOwl** [comment]
The only part I am not sure about is the ability to rename your job titles. I can see that very quickly being abused. I feel that should very much still be something that is controlled by command staff only and not in the general crew hands.

**Comment by lessthnthree** [comment]
Someone brought up the very good point of antags. Removing job titles from the scope of this, it needs more time and planning if ever.

File: code/modules/modular_computers/computers/item/pda.dm [hunk 1]
```diff
@@ -40,6 +40,7 @@
 		/datum/computer_file/program/notepad,
 		// NOVA EDIT ADDITION START
 		/datum/computer_file/program/crew_manifest, // Adds crew manifest to all base tablets
+		/datum/computer_file/program/crew_self_serve, // Adds crew self serve to all base tablets
 		/datum/computer_file/program/maintenance/camera // Adds camera to all base tablets
 		// SKRAT EDIT ADDITION END
```
Comment (FlufflesTheDog) [inline code/modules/modular_computers/computers/item/pda.dm:45]:
```suggestion
		// NOVA EDIT ADDITION END
```
lol

Comment (lessthnthree) [reply code/modules/modular_computers/computers/item/pda.dm]: lol.
copy pasting shit for 3 PRs on 3 codebases is a constant kick in the crotch

**Pull request closed without merging**